src/static_invoice_store.rs

```diff
@@ -0,0 +1,71 @@
+use crate::{hex_utils, rate_limiter::RateLimiter, types::DynStore};
+use bitcoin::hashes::{sha256, Hash};
+use lightning::{offers::static_invoice::StaticInvoice, util::ser::Writeable};
+use std::{
+	sync::{Arc, Mutex},
+	time::Duration,
+};
+
+pub(crate) struct StaticInvoiceStore {
+	kv_store: Arc<DynStore>,
+	rate_limiter: Mutex<RateLimiter>,
```

Comment:
Needed to add `Mutex` here even though I don't think it is used concurrently. Otherwise `handle_event` needed to be made mutable, the Arc higher up no longer working, etc.